packages/cmrjs/src/cmr-utils.js

```diff
@@ -203,43 +211,55 @@ async function getXMLMetadataAsString(xmlFilePath) {
  * @returns {Promise<Object>} promise resolves to object version of the xml
  */
 async function parseXmlString(xml) {
-  return (promisify(xml2js.parseString))(xml, xmlParseOptions);
+  return promisify(xml2js.parseString)(xml, xmlParseOptions);
 }
 
 /**
- * return UMMG metadata object from CMR UMM-G json file
- * @param {string} cmrFilename - s3 path to json file
+ * Returns UMMG metadata object from CMR UMM-G JSON file in S3.
+ *
+ * @param {string} cmrFilename - S3 path to JSON file
+ * @param {string} [etag] - optional entity tag for the desired version of the
+ *    CMR file
  * @returns {Promise<Object>} CMR UMMG metadata object
  */
-async function metadataObjectFromCMRJSONFile(cmrFilename) {
+async function metadataObjectFromCMRJSONFile(cmrFilename, etag) {
   const { Bucket, Key } = parseS3Uri(cmrFilename);
-  const obj = await getS3Object(Bucket, Key, { retries: 5 });
+  const obj = await getS3Object({ Bucket, Key, IfMatch: etag }, { retries: 5 });
   return JSON.parse(obj.Body.toString());
 }
 
 /**
- * return metadata object from cmr echo10 XML file.
- * @param {string} cmrFilename
- * @returns {Promise<Object>} cmr xml metadata as object.
+ * Returns metadata object from CMR Echo10 XML file in S3.
+ *
+ * @param {string} cmrFilename - S3 path to XML file
+ * @param {string} [etag] - optional entity tag for the desired version of the
+ *    CMR file
+ * @returns {Promise<Object>} CMR XML metadata object
  */
-async function metadataObjectFromCMRXMLFile(cmrFilename) {
-  const metadata = await getXMLMetadataAsString(cmrFilename);
-  return parseXmlString(metadata);
-}
+const metadataObjectFromCMRXMLFile = (cmrFilename, etag) =>
+  getXMLMetadataAsString(cmrFilename, etag).then(parseXmlString);
 
 /**
- * Return cmr metadata object from a CMR Echo10XML file or CMR UMMG File.
- * @param {string} cmrFilename - s3 path to cmr file
- * @returns {Promise<Object>} - metadata object from the file.
+ * Returns CMR metadata object from a CMR ECHO-10 XML file or CMR UMMG JSON
+ * file in S3.
+ *
+ * @param {string} cmrFilename - S3 path to CMR file
+ * @param {string} [etag] - optional entity tag for the desired version of the
+ *    CMR file
+ * @returns {Promise<Object>} metadata object from the file
+ * @throws {Error} if the specified filename does not represent an ECHO-10 XML
+ *    file or a UMMG file
+ * @see isECHO10File
+ * @see isUMMGFile
  */
-async function metadataObjectFromCMRFile(cmrFilename) {
+function metadataObjectFromCMRFile(cmrFilename, etag) {
   if (isECHO10File(cmrFilename)) {
-    return metadataObjectFromCMRXMLFile(cmrFilename);
+    return metadataObjectFromCMRXMLFile(cmrFilename, etag);
   }
   if (isUMMGFile(cmrFilename)) {
-    return metadataObjectFromCMRJSONFile(cmrFilename);
+    return metadataObjectFromCMRJSONFile(cmrFilename, etag);
   }
-  throw new Error(`cannot return metdata from invalid cmrFilename: ${cmrFilename}`);
+  throw new Error(`Invalid CMR filename: ${cmrFilename}`);
```

Comment:
I personally think the previous error message was a little more descriptive, but it's not a huge deal